### PR TITLE
feat(ga/webpagetest): add webpagetest workflow

### PR DESCRIPTION
**Pull request by arnostpleskot**
WebPageTest + Get Netlify Deploy action: more info here: https://github.com/strvcom/strv-io-web/pull/126#issuecomment-621915771

### CODE REVIEW

**Comment by arnostpleskot** [comment]
Hello @dannytce @koss-lebedev @HonzaBartos!

I've finally put it together this two actions:
- `.github/actions/get-netlify-preview/`
  - this one will ping netlify API for latest deploy, wait until it's ready and then outputs its URL
- `.github/actions/run-web-page-test/`
  - this will trigger webpagetest and then prints report as a comment

There will be several next steps but I want to sync that this is direction that we will want to take before creating new repo:
- create repos for both actions
- add readme
- test with different scenarios than pull request
- publish on GitHub marketplace

One thing I'm not happy about and I want to investigate in the future is the way how netlify action is getting build - right now we are simply getting latest build, this is okeysh solution for most cases, but could cause problems in the case when there are several builds for different branches in the same time. The solution is simple - find build by the commit, which is possible, but in the case of PR there is in [github action context](https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions) PR that is created as merge commit between branches (which is not triggering netlify deploy).

**Comment by dannytce** [comment]
Great stuff @arnostpleskot! Thank you for putting this together :) 

A few ideas about how to improve it:
1. Let's wrap the webpagetest output in the collapsible view. The comment output is very long.
2. Maybe let's print some emojis if any of the values is too high?
3. Come up with an idea, how do to detect regressions? E.g. some value changed from previous deploy. So it would be cool to highlight the change and write down, what's changed and how.

When it comes to the next steps. I think those could be done separately in a new PR :) 

Btw have you seen following actions?
https://github.com/JakePartusch/wait-for-netlify-action
https://github.com/JCofman/webPagetestAction

Would be cool to know the reasons why you wrote own actions. I think this could be reflected in the action's READMEs :)

**Comment by arnostpleskot** [comment]
@dannytce thank you for feedback. 
1. Folding is definitely a great idea - but all the data showed here are also shown on webpagetest.org, so I'm not sure if it just isn't better to go straight there - maybe we can fold those waterfall charts.
2. Then we need to set some thresholds and defaults to decide what values are high. Also, webpagetest provides some grading, but I wasn't able to find it in results (it is 50 000+ lines long JSON).
3. Regression is a good idea as well - I need to check if it is possible to use [github actions cache](https://github.com/actions/toolkit/tree/master/packages/tool-cache#actionstool-cache) to save results between runs.

About those two actions I've been testing them in the beginning but they are useless:
- https://github.com/JakePartusch/wait-for-netlify-action is dumb and is just polling (with axios 👎) `https://deploy-preview-${PR_NUMBER}--${siteName}.netlify.app` which will give false results when you have more than 1 commit in PR. It will just trigger any following step for a previously deployed page not for deploy created by the related commit.
- https://github.com/JCofman/webPagetestAction - doesn't work with push requests and code is pretty ugly - also IMHO it is printing a lot of useless info.

File: .github/actions/run-web-page-test/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "Arnost Pleskot <me@pleskot.net>",
+  "license": "ISC",
+  "dependencies": {
+    "@actions/core": "^1.2.3",
+    "@actions/github": "^2.1.1",
+    "got": "^11.0.2",
```
Comment (dannytce) [inline .github/actions/run-web-page-test/package.json:14]: This one is not needed, isn't it? :)

**Pull request closed without merging**